src/parsers/manifestjson.js

```diff
@@ -40,6 +40,20 @@ import {
 } from 'utils';
 import BLOCKED_CONTENT_SCRIPT_HOSTS from 'blocked_content_script_hosts.txt';
 
+async function getStreamImageSize(stream) {
+  const chunks = [];
+  for await (const chunk of stream) {
+    chunks.push(chunk);
+    try {
+      return getImageSize(Buffer.concat(chunks));
+    } catch (error) {
+      /* Not ready yet */
```

Comment:
What does that mean exactly? Is it that we don't have enough data to detect the image size?

Reply:
That's correct. Unfortunately `getImageSize` will throw an error when it can't determine the size of the file. In this case, it happens repeatedly until it receives enough information.

My guess though is that this information is within the first chunk, at least that's what I experienced in my few tests with the `http` module.

Should it be

```suggestion
      /* The size information isn't available yet */
```

Reply:
> Should it be

yeah